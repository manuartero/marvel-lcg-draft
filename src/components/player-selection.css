#color-picker-container {
  background-color: var(--medium-dark-blue);
  border: 1px solid rgb(38, 38, 39);
  color: white;
}

.faction-claim-Aggression {
  color: rgb(239 68 68);
}

.faction-claim-Leadership {
  color: rgb(59 130 246);
}

.faction-claim-Justice {
  color: rgb(234 179 8);
}

.faction-claim-Protection {
  color: rgb(34 197 94);
}

.color-box {
  opacity: 0.5;
  width: 100px;
  height: 100px;
  border: 4px solid rgb(243, 243, 243);
  border-radius: 4px;
  transition: 0.25s;
}

.color-box:hover {
  opacity: 1;
}

.color-box.selected {
  opacity: 1;
}

.hero-min {
  background-position: initial;
}
