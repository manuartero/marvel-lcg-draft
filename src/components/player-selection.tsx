import c from "classnames";
import { useState } from "react";
import { ReadyButton } from "elements/ready-button";

import type { Selection } from "app-domain";
import type { CardFaction } from "services/cards";

import "./player-selection.css";

type Props = {
  onReady: (selection: Selection<CardFaction>) => void;
};

export function PlayerSelection({ onReady }: Props) {
  const [player1Faction, setPlayer1Faction] = useState<CardFaction>();
  const [player2Faction, setPlayer2Faction] = useState<CardFaction>();

  const isReady = player1Faction && player2Faction;

  const handleReady = () => {
    if (isReady) {
      onReady({ player1: player1Faction, player2: player2Faction });
    }
  };

  return (
    <section className="h-screen w-screen flex flex-col justify-center items-center">
      <PlayerFaction
        faction={player1Faction}
        title="Player 1"
        setter={setPlayer1Faction}
      />
      <PlayerFaction
        faction={player2Faction}
        title="Player 2"
        setter={setPlayer2Faction}
      />
      <div className="flex justify-center mt-4">
        <ReadyButton disabled={!isReady} onClick={handleReady} />
      </div>
    </section>
  );
}

type PlayerFactionProps = {
  title: string;
  setter: React.Dispatch<React.SetStateAction<CardFaction | undefined>>;
  faction: CardFaction | undefined;
};

function PlayerFaction({ title, setter, faction }: PlayerFactionProps) {
  return (
    <article
      id="color-picker-container"
      className="w-1/2 bg-white p-4 rounded-lg shadow-lg mt-4"
    >
      <div className="flex items-center justify-between mb-4">
        <div>
          <h2 className="text-2xl font-semibold me-2">{title}</h2>
        </div>
        <div>
          {faction && (
            <h3 className={c(`faction-claim-${faction}`, "mr-8")}>{faction}</h3>
          )}
        </div>
      </div>

      <section className="flex justify-around mt-4 mb-4" role="group">
        <div
          className={`color-box bg-red-500 cursor-pointer ${
            faction === "Aggression" && "selected"
          }`}
          aria-label="Aggression Faction"
          onClick={() => {
            setter("Aggression");
          }}
        ></div>
        <div
          className={`color-box bg-blue-500 cursor-pointer ${
            faction === "Leadership" && "selected"
          }`}
          aria-label="Leadership Faction"
          onClick={() => {
            setter("Leadership");
          }}
        ></div>
        <div
          className={`color-box bg-yellow-500 cursor-pointer ${
            faction === "Justice" && "selected"
          }`}
          aria-label="Justice Faction"
          onClick={() => {
            setter("Justice");
          }}
        ></div>
        <div
          className={`color-box bg-green-500 cursor-pointer ${
            faction === "Protection" && "selected"
          }`}
          aria-label="Protection Faction"
          onClick={() => {
            setter("Protection");
          }}
        ></div>
      </section>
    </article>
  );
}
