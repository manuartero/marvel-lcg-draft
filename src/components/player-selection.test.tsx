import { render } from "@testing-library/react";
import { PlayerSelection } from "./player-selection";

describe("<PlayerSelection />", () => {
  test("renders", () => {
    const { asFragment } = render(<PlayerSelection onReady={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test.todo("calls onReady() when ready button is clicked");
});
