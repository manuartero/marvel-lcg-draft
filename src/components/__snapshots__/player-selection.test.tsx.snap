// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerSelection /> renders 1`] = `
<DocumentFragment>
  <section
    class="h-screen w-screen flex flex-col justify-center items-center"
  >
    <article
      class="w-1/2 bg-white p-4 rounded-lg shadow-lg mt-4"
      id="color-picker-container"
    >
      <div
        class="flex items-center justify-between mb-4"
      >
        <div>
          <h2
            class="text-2xl font-semibold me-2"
          >
            Player 1
          </h2>
        </div>
        <div />
      </div>
      <section
        class="flex justify-around mt-4 mb-4"
        role="group"
      >
        <div
          aria-label="Aggression Faction"
          class="color-box bg-red-500 cursor-pointer false"
        />
        <div
          aria-label="Leadership Faction"
          class="color-box bg-blue-500 cursor-pointer false"
        />
        <div
          aria-label="Justice Faction"
          class="color-box bg-yellow-500 cursor-pointer false"
        />
        <div
          aria-label="Protection Faction"
          class="color-box bg-green-500 cursor-pointer false"
        />
      </section>
    </article>
    <article
      class="w-1/2 bg-white p-4 rounded-lg shadow-lg mt-4"
      id="color-picker-container"
    >
      <div
        class="flex items-center justify-between mb-4"
      >
        <div>
          <h2
            class="text-2xl font-semibold me-2"
          >
            Player 2
          </h2>
        </div>
        <div />
      </div>
      <section
        class="flex justify-around mt-4 mb-4"
        role="group"
      >
        <div
          aria-label="Aggression Faction"
          class="color-box bg-red-500 cursor-pointer false"
        />
        <div
          aria-label="Leadership Faction"
          class="color-box bg-blue-500 cursor-pointer false"
        />
        <div
          aria-label="Justice Faction"
          class="color-box bg-yellow-500 cursor-pointer false"
        />
        <div
          aria-label="Protection Faction"
          class="color-box bg-green-500 cursor-pointer false"
        />
      </section>
    </article>
    <div
      class="flex justify-center mt-4"
    >
      <button
        class="px-10 py-4 text-white font-semibold bg-sky-950  hover:bg-sky-700 rounded-md opacity-50 cursor-not-allowed"
        disabled=""
      >
        READY!
      </button>
    </div>
  </section>
</DocumentFragment>
`;
